staging/src/k8s.io/apiserver/pkg/authentication/request/bearertoken/bearertoken.go

```diff
@@ -58,6 +58,10 @@ func (a *Authenticator) AuthenticateRequest(req *http.Request) (*authenticator.R
 		req.Header.Del("Authorization")
 	}
 
+	if resp != nil {
+		resp.AuthenticatorName = "bearer-token"
```

Comment:
Not really a fan of making all of these names into APIs :(

At least our metrics are declared as alpha ...

Reply:
Open to suggestions here.

I have some very concrete business requirements to classify auth metrics. Examples,
1. Justify deprecation of certain auth types based on usage.
2. Route alerts to different teams based on the authenticator type.

Reply:
this overrides any existing AuthenticatorName in the resp... I was expecting to be able to distinguish between webhook token, legacy serviceaccount token, bound serviceaccount token, tokenfile, and loopback apiserver token authentication

Reply:
it seems reasonable to set this to token or bearer-token if resp is non-nil and has no AuthenticatorName set

Reply:
also see [comment](https://github.com/kubernetes/kubernetes/pull/85113/files#r354888449) about separating auth method and authenticator id

Reply:
The first two comments in this block are resolved.